### PR TITLE
removes some DRY

### PR DESCRIPTION
**Pull request by stoeffel**
Needs some review...

### CODE REVIEW

File: index.es6 [hunk 1]
```diff
@@ -1,22 +1,18 @@
...
 }
 
 function _hasClass(getClass, elements) {
   return elements.every((element, idx) => element.classList.contains(getClass(idx)));
 }
 
-export let addClass = curry2(_addClass);
-export let removeClass = curry2(_removeClass);
-export let toggleClass = curry2(_toggleClass);
+let ClassList = DOMTokenList.prototype;
```
Comment (hemanth) [inline index.es6:12]: :+1:

File: index.es6 [hunk 2]
```diff
@@ -1,22 +1,18 @@
...
-  return elements.map((element, idx) => element.classList.add(getClass(idx)));
-}
-
-function _removeClass(getClass, elements) {
-  return elements.map((element, idx) => element.classList.remove(getClass(idx)));
-}
 
-function _toggleClass(getClass, elements) {
-  return elements.map((element, idx) => element.classList.toggle(getClass(idx)));
+function recur(modify, getClass, elements) {
```
Comment (hemanth) [inline index.es6]: Can we use arrow function here? Looking for a better name than `recur`...hmm

**Comment by stoeffel** [comment]
How about. `modifyElements` instead of `recur`?

**Comment by stoeffel** [comment]
I think it's ready to :ship: now.

**Comment by hemanth** [comment]
@stoeffel :shipit:

**Pull request merged**